test/ig/IGExporter.IG.test.ts

```diff
@@ -0,0 +1,548 @@
+                  nameUrl: 'unsupported.html',
+                  title: 'Unsupported',
+                  generation: 'html'
+                }
+              ]
+            }
+          ]
+        }
+      ];
+      exporter.export(tempOut);
+      const igPath = path.join(tempOut, 'input', 'ImplementationGuide-sushi-test.json');
+      expect(fs.existsSync(igPath)).toBeTruthy();
+      const igContent = fs.readJSONSync(igPath);
+      expect(igContent.definition.page.page).toEqual([
+        {
+          nameUrl: 'index.html',
+          title: 'Home',
+          generation: 'markdown',
+          page: [
+            {
+              nameUrl: 'something-special.html',
+              title: 'Something Special',
+              generation: 'xml'
+            },
+            {
+              nameUrl: 'other-page.html',
+              title: 'Other Page',
+              generation: 'markdown',
+              page: [
+                {
+                  nameUrl: 'unsupported.html',
+                  title: 'Unsupported',
+                  generation: 'html'
+                }
+              ]
+            }
+          ]
+        }
+      ]);
+    });
+
+    it('should not include a page attribute when all subpages are intro or notes files', () => {
+      config.pages = [
+        {
+          nameUrl: 'index.md',
+          title: 'Home',
+          generation: 'markdown',
+          page: [
+            {
+              nameUrl: 'main-intro.md'
+            },
+            {
+              nameUrl: 'main-notes.md'
+            }
+          ]
+        }
+      ];
+      exporter.export(tempOut);
+      const igPath = path.join(tempOut, 'input', 'ImplementationGuide-sushi-test.json');
+      expect(fs.existsSync(igPath)).toBeTruthy();
+      const igContent = fs.readJSONSync(igPath);
+      expect(igContent.definition.page.page).toEqual([
+        {
+          nameUrl: 'index.html',
+          title: 'Home',
+          generation: 'markdown'
+        }
+      ]);
+    });
+
+    it('should log an error when no file exists for a configured page', () => {
+      config.pages = [
+        {
+          nameUrl: 'index.md',
+          title: 'Home',
+          generation: 'markdown'
+        },
+        {
+          nameUrl: 'nothing.md',
+          title: 'Nothing',
+          generation: 'markdown'
+        }
+      ];
+      exporter.export(tempOut);
+      const igPath = path.join(tempOut, 'input', 'ImplementationGuide-sushi-test.json');
+      expect(fs.existsSync(igPath)).toBeTruthy();
+      const igContent = fs.readJSONSync(igPath);
+      expect(igContent.definition.page.page).toEqual([
+        {
+          nameUrl: 'index.html',
+          title: 'Home',
+          generation: 'markdown'
+        },
+        {
+          nameUrl: 'nothing.html',
+          title: 'Nothing',
+          generation: 'markdown'
+        }
+      ]);
+      expect(loggerSpy.getLastMessage('error')).toMatch(/nothing\.md not found/s);
+    });
+  });
```

Comment:
I feel like there should be more tests around `config.resources` -- as these tests only cover when the config doesn't specify resources.  But we need to test use cases like omitted resources, partial resource info, etc.

Reply:
We also need tests for `config.groups` when you've implemented that.